omnibus/config/software/datadog-agent-finalize.rb

```diff
@@ -14,41 +14,47 @@
-        #move "#{install_dir}/agent/checks.d", conf_dir
-        delete "#{install_dir}/bin/agent/agent.exe"
-        # TODO why does this get generated at all
-        delete "#{install_dir}/bin/agent/agent.exe~"
-    else
-        # Move checks and configuration files
-        mkdir "/etc/datadog-agent"
-        move "#{install_dir}/bin/agent/dd-agent", "/usr/bin/dd-agent"
-        move "#{install_dir}/etc/datadog-agent/datadog.yaml.example", "/etc/datadog-agent"
-        move "#{install_dir}/etc/datadog-agent/trace-agent.conf", "/etc/datadog-agent/trace-agent.conf.example"
-        move "#{install_dir}/etc/datadog-agent/process-agent.conf", "/etc/datadog-agent/process-agent.conf.example"
-        move "#{install_dir}/etc/datadog-agent/conf.d", "/etc/datadog-agent", :force=>true
-        move "#{install_dir}/bin/agent/dist/conf.d/*", "/etc/datadog-agent/conf.d"
-        move "#{install_dir}/agent/checks.d", "#{install_dir}/checks.d"
+    # TODO too many things done here, should be split
+    block do
+        if windows?
+            conf_dir_root = "#{Omnibus::Config.source_dir()}/etc/datadog-agent"
+            conf_dir = "#{conf_dir_root}/extra_package_files/EXAMPLECONFSLOCATION"
+            mkdir conf_dir
+            move "#{install_dir}/etc/datadog-agent/datadog.yaml.example", conf_dir_root, :force=>true
+            move "#{install_dir}/etc/datadog-agent/trace-agent.conf", conf_dir_root, :force=>true
+            #move "#{install_dir}/etc/datadog-agent/process-agent.conf", conf_dir
+            move "#{install_dir}/etc/datadog-agent/conf.d/*", conf_dir, :force=>true
+            #move "#{install_dir}/agent/checks.d", conf_dir
+            delete "#{install_dir}/bin/agent/agent.exe"
+            # TODO why does this get generated at all
+            delete "#{install_dir}/bin/agent/agent.exe~"
+        else
+            # Move checks and configuration files
+            mkdir "/etc/datadog-agent"
+            move "#{install_dir}/bin/agent/dd-agent", "/usr/bin/dd-agent"
+            move "#{install_dir}/etc/datadog-agent/datadog.yaml.example", "/etc/datadog-agent"
+            move "#{install_dir}/etc/datadog-agent/trace-agent.conf", "/etc/datadog-agent/trace-agent.conf.example"
+            move "#{install_dir}/etc/datadog-agent/process-agent.conf", "/etc/datadog-agent/process-agent.conf.example"
+            move "#{install_dir}/etc/datadog-agent/conf.d", "/etc/datadog-agent", :force=>true
+            move "#{install_dir}/bin/agent/dist/conf.d/*.yaml*", "/etc/datadog-agent/conf.d/"
+            Dir.glob("#{install_dir}/bin/agent/dist/conf.d/**/*.d").each do |check_dir|
+                dir_name = File.basename check_dir
+                mkdir "/etc/datadog-agent/conf.d/#{dir_name}" unless File.exists? "/etc/datadog-agent/conf.d/#{dir_name}"
+                move "#{install_dir}/bin/agent/dist/conf.d/#{dir_name}/*.yaml*", "/etc/datadog-agent/conf.d/#{dir_name}/", :force=>true
+            end
```

Comment:
why do we have to do this complicated stuff instead of just `move`-ing the `conf.d` directory to its correct install location?

Reply:
The problem is that at this point, the `conf.d` is already existing with integration configuration in it so we can't replace it.
We might be able to do something smarter by reordering some build orders though, I think @masci is planning to revamp a bit the build definitions

Reply:
👍  thanks for the explanation!

Could you add a small inline comment here to explain this? :)